src/posix/platform/spinel_interface.hpp

```diff
@@ -0,0 +1,180 @@
+    }
+
+    /**
+     * This method gets the `RxFrameBuffer`.
+     *
+     * The receive frame buffer is an `Hdlc::MultiFrameBuffer` and therefore it is capable of storing multiple
+     * spinel frames in a FIFO queue manner. The `RxFrameBuffer` contains the received spinel frames.
+     *
+     * When during `Process()` or `WaitForFrame()` the `Callbacks::HandleReceivedFrame()` is invoked, the newly
+     * received spinel frame is available in the receive frame buffer. The callback is expected to either process
+     * and then discard the frame (using `RxFrameBuffer::DiscardFrame()` method) or save the frame
+     * (using `RxFrameBuffer::SaveFrame()` so that it can be read later.
+     *
+     * @returns A reference to receive frame buffer containing newly received spinel frame or previously saved spinel
+     *          frames.
+     *
+     */
+    RxFrameBuffer &GetRxFrameBuffer(void) { return mRxFrameBuffer; }
+
+    /**
+     * This method initializes the interface to the Radio Co-processor (RCP)
+     *
+     * @note This method should be called before reading and sending spinel frames to the interface.
+     *
+     * @param[in]   aConfig  Parameters to be given to the device or executable.
+     *
+     * @retval OT_ERROR_NONE          The interface is initialized successfully
+     * @retval OT_ERROR_ALREADY       The interface is already initialized.
+     * @retval OT_ERROR_INVALID_ARGS  The device or executable cannot be found or failed to open/run.
+     *
+     */
+    virtual otError Init(const char *aRadioFile, const char *aRadioConfig) = 0;
+
+    /**
+     * This method deinitializes the interface to the Radio Co-processor (RCP).
+     *
+     */
+    virtual void Deinit(void) = 0;
+
+    /**
+     * This method encodes and sends a spinel frame to Radio Co-processor (RCP) over the socket.
+     *
+     * This is blocking call, i.e., if the socket is not writable, this method waits for it to become writable for
+     * up to `kMaxWaitTime` interval.
+     *
+     * @param[in] aFrame     A pointer to buffer containing the spinel frame to send.
+     * @param[in] aLength    The length (number of bytes) in the frame.
+     *
+     * @retval OT_ERROR_NONE     Successfully encoded and sent the spinel frame.
+     * @retval OT_ERROR_NO_BUFS  Insufficient buffer space available to encode the frame.
+     * @retval OT_ERROR_FAILED   Failed to send due to socket not becoming writable within `kMaxWaitTime`.
+     *
+     */
+    virtual otError SendFrame(const uint8_t *aFrame, uint16_t aLength) = 0;
+
+    /**
+     * This method waits for receiving part or all of spinel frame within specified interval.
+     *
+     * @param[in]  aTimeout  A reference to the timeout.
+     *
+     * @retval OT_ERROR_NONE             Part or all of spinel frame is received.
+     * @retval OT_ERROR_RESPONSE_TIMEOUT No spinel frame is received within @p aTimeout.
+     *
+     */
+    virtual otError WaitForFrame(struct timeval &aTimeout) = 0;
```

Comment:
Add `const` to `aTimeout`?

Reply:
Lower layer uses `aTimeout` as a parameter of the function `select()`. The type of the parameter is ` struct timeval *timeout` in function `select()`.  If we add `const` to `aTimeout`, the compiler will output errors.